### DISCUSSION TITLE
Error: Cannot find module '.prisma/client/index'

### DISCUSSION THREAD

**Discussion by anushthakumari** [prisma/prisma#21422] [Q&A]
### Bug description

I am using smarterasp.net for deploying my next application. I have to zip entire app(including node_modules) and extract it.

I referred following official guide-
[link](https://www.smarterasp.net/support/kb/a2233/how-to-publish-a-next_js-project-to-your-hosting-account.aspx)

I tried deploying my NextJs application, locally its working fine however when I deploy it, it shows follwing error-

Error: Cannot find module '.prisma/client/index'
Require stack:
- h:\root\home\[dir]\www\site1\node_modules\@prisma\client\index.js
- h:\root\home\[dir]\www\site1\.next\server\app\page.js
- h:\root\home\[dir]\www\site1\node_modules\next\dist\server\require.js
- h:\root\home\[dir]\www\site1\node_modules\next\dist\server\load-components.js

### Expected behavior

i should render the page/

### Prisma information

```prisma
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model admins {
  id        Int      @id @default(autoincrement())
  email     String   @unique
  pass      String
  createdAt DateTime @default(now())
}
```

```ts
import { PrismaClient } from "@prisma/client";

const prismaClientSingleton = () => {
	return new PrismaClient();
};

const globalForPrisma = {
	prisma: undefined,
};

const prisma = globalForPrisma.prisma ?? prismaClientSingleton();

export default prisma;

if (process.env.NODE_ENV !== "production") globalForPrisma.prisma = prisma;

```

### Environment & setup

- OS: windows
- Database: MySQL 8.*.*
- Node.js version: 18.17.1

### Prisma Version

```
5.3.1
```

**Comment by Eagleto** [comment]
Ensure that the Node.js version you are using locally for development matches the Node.js version on the server. Some packages might be sensitive to the Node.js version, and using different versions can lead to compatibility issues.
Verify that you are correctly zipping your entire application, including the node_modules folder, as you mentioned.
Make sure that the versions of your dependencies, including Prisma and Next.js, are consistent between your local environment and the deployment environment.

**Reply by ludralph** [reply]
Hi @anushthakumari 👋 

Did you get a chance to see Eagleto response and do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by anushthakumari** [reply]
Hii @ludralph , I tried Eagleto's answer but that didn't help.

**Reply by ludralph** [reply]
Hi @anushthakumari, 

The error message you're seeing indicates that the Prisma Client is not being found in your deployment. It's important to ensure that the Prisma Client is generated during the build process. Add `prisma generate` to the `postinstall` script of your application: Here's how you can do this in your package.json file:
```json
{
  "scripts": {
    "postinstall": "prisma generate"
  }
}
```
Let me know if this helps.

**Reply by anushthakumari** [reply]
I already have this scripts in my package.json

**Reply by Eagleto** [reply]
Share your package.json if you can

**Reply by Eagleto** [reply]
Add this things in scripts in package.json

    "predev": "prisma generate && prisma migrate deploy",    "setup": "prisma generate && prisma migrate deploy",    "prisma": "prisma"

**Reply by anushthakumari** [reply]
I will try, this project of mine is on hold, I will update u once it gets unhold

**Reply by ashburnham** [reply]
Any update? I'm getting the same.

It works on my machine (mac M1) but then this happens when deploying on vercel...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by manojtsx** [reply]
This issue hasn't been resolved mine case
